cuda_memtest.cu

```diff
@@ -76,6 +76,7 @@ pthread_mutex_t atomic_mutex = PTHREAD_MUTEX_INITIALIZER;
 void run_tests(char*, unsigned int);
 extern void update_temperature(void);
 extern void allocate_small_mem(void);
+volatile int active_update_temperature;
```

Comment:
It is read by different threads and the main thread might change it.

basically, it allows for clean shut-downs of threads, because I need a
return type and a criteria to avoid an infinite loop (both cause
warnings ^^).